lib/HotModuleReplacementPlugin.js

```diff
@@ -3,260 +3,249 @@
+							hotUpdateMainContent.c[chunkId] = true;
+							currentChunk.files.push(filename);
+							this.applyPlugins("chunk-asset", currentChunk, filename);
+						}
+					} else {
+						hotUpdateMainContent.c[chunkId] = false;
 					}
-				} else {
-					hotUpdateMainContent.c[chunkId] = false;
-				}
-			}, this);
-			var source = new RawSource(JSON.stringify(hotUpdateMainContent));
-			var filename = this.getPath(hotUpdateMainFilename, {
-				hash: records.hash
+				}, this);
+				const source = new RawSource(JSON.stringify(hotUpdateMainContent));
+				const filename = this.getPath(hotUpdateMainFilename, {
+					hash: records.hash
+				});
+				this.assets[filename] = source;
 			});
-			this.assets[filename] = source;
-		});
 
-		compilation.mainTemplate.plugin("hash", function(hash) {
-			hash.update("HotMainTemplateDecorator");
-		});
-
-		compilation.mainTemplate.plugin("module-require", function(_, chunk, hash, varModuleId) {
-			return "hotCreateRequire(" + varModuleId + ")";
-		});
+			compilation.mainTemplate.plugin("hash", hash => {
+				hash.update("HotMainTemplateDecorator");
+			});
 
-		compilation.mainTemplate.plugin("require-extensions", function(source) {
-			var buf = [source];
-			buf.push("");
-			buf.push("// __webpack_hash__");
-			buf.push(this.requireFn + ".h = function() { return hotCurrentHash; };");
-			return this.asString(buf);
-		});
+			compilation.mainTemplate.plugin("module-require", (_, chunk, hash, varModuleId) => {
+				return `hotCreateRequire(${varModuleId})`;
+			});
 
-		compilation.mainTemplate.plugin("bootstrap", function(source, chunk, hash) {
-			source = this.applyPluginsWaterfall("hot-bootstrap", source, chunk, hash);
-			return this.asString([
-				source,
-				"",
-				hotInitCode
-				.replace(/\$require\$/g, this.requireFn)
-				.replace(/\$hash\$/g, JSON.stringify(hash))
-				.replace(/\$requestTimeout\$/g, requestTimeout)
-				.replace(/\/\*foreachInstalledChunks\*\//g, chunk.chunks.length > 0 ? "for(var chunkId in installedChunks)" : "var chunkId = " + JSON.stringify(chunk.id) + ";")
-			]);
-		});
+			compilation.mainTemplate.plugin("require-extensions", function(source) {
+				const buf = [source];
+				buf.push("");
+				buf.push("// __webpack_hash__");
+				buf.push(`${this.requireFn}.h = () => hotCurrentHash`);
+				return this.asString(buf);
+			});
 
-		compilation.mainTemplate.plugin("global-hash", function() {
-			return true;
-		});
+			compilation.mainTemplate.plugin("bootstrap", function(source, chunk, hash) {
+				source = this.applyPluginsWaterfall("hot-bootstrap", source, chunk, hash);
+				return this.asString([
+					source,
+					"",
+					hotInitCode
+					.replace(/\$require\$/g, this.requireFn)
+					.replace(/\$hash\$/g, JSON.stringify(hash))
+					.replace(/\$requestTimeout\$/g, requestTimeout)
+					.replace(/\/\*foreachInstalledChunks\*\//g, chunk.chunks.length > 0 ? "for(const chunkId in installedChunks)" : `const chunkId = ${JSON.stringify(chunk.id)};`)
```

Comment:
`const chunkId` -> `var chunkId`

This is code which is in the generated bundle. The generated code should not contain higher ES.